### PR TITLE
ETK Performance: Defer loading the starter page templates plugins

### PR DESCRIPTION
**Pull request by arthur791004**
Related to pdDR7T-12w-p2

## Proposed Changes

* Defer loading the starter page templates plugins
  * starter-page-templates
  * starter-page-templates-error

## Testing Instructions

1. Checkout this branch.
2. in apps/editing-toolkit, run `yarn dev --sync`.
3. Create a new page on your sandboxed simple site
4. Make sure the starter page templates works as expected

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/class-starter-page-templates.php [hunk 1]
```diff
@@ -85,7 +85,9 @@ public function register_scripts() {
 			plugins_url( 'dist/starter-page-templates.min.js', __FILE__ ),
 			array( 'wp-plugins', 'wp-edit-post', 'wp-element' ),
 			filemtime( plugin_dir_path( __FILE__ ) . 'dist/starter-page-templates.min.js' ),
-			true
+			array(
+				'strategy' => 'defer',
+			)
```
Comment (alshakero) [inline apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/class-starter-page-templates.php]: Turns out there is a backwards compatible way that goes like this:

```suggestion
 	true 
 );
 		wp_script_add_data( 'starter-page-templates', 'strategy', 'defer' );

```

File: apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/class-starter-page-templates.php [hunk 2]
```diff
@@ -164,7 +166,9 @@ public function pass_error_to_frontend( $message ) {
 			null,
 			array(),
 			'1.O',
-			true
+			array(
+				'strategy' => 'defer',
+			)
```
Comment (alshakero) [inline apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/class-starter-page-templates.php]: +1

**Review by alshakero — CHANGES_REQUESTED** [review]

**Review by alshakero — APPROVED** [review]
Code looks great 👍🏼

**Comment by arthur791004** [comment]
Closing it as it's not planned. See pdDR7T-12w-p2#comment-1329.

**Pull request closed without merging**